{
  try {
    ready=true;
    return qp.next();
  }
 catch (  Exception ex) {
    throw new BaseXException(ex);
  }
}
